tests/integration/MyObservations.test.js

```diff
@@ -125,6 +127,35 @@ describe( "MyObservations", ( ) => {
         } )
       ];
 
+      // Mock inatjs endpoints so they return the right responses for the right test data
+      inatjs.observations.create.mockImplementation( ( params, _opts ) => {
+        const mockObs = mockObservations.find( o => o.uuid === params.observation.uuid );
+        return Promise.resolve( makeResponse( [{ id: faker.number.int( ), uuid: mockObs.uuid }] ) );
+      } );
+      inatjs.observation_photos.create.mockImplementation( async ( params, _opts ) => {
+        const mockObsPhotos = flatten( mockObservations.map( o => o.observationPhotos ) );
+        const mockObsPhoto = mockObsPhotos.find(
+          op => op.photo.id === params.observation_photo.photo_id
+        );
+        // Pretend this takes a bit
+        await sleep( 500 );
+        return makeResponse( [{
+          id: faker.number.int( ),
+          uuid: mockObsPhoto.uuid
+        }] );
+      } );
+      inatjs.photos.create.mockImplementation( ( params, _opts ) => {
+        const mockObsPhotos = flatten( mockObservations.map( o => o.observationPhotos ) );
+        const mockPhotos = mockObsPhotos.map( op => op.photo );
+        const mockPhoto = mockPhotos.find( p => p.id === params.photo.id );
+        // Pretend this takes a bit
+        // await sleep( 200 );
+        return Promise.resolve( makeResponse( [{
+          id: faker.number.int( ),
+          uuid: mockPhoto.uuid
+        }] ) );
+      } );
+
```

Comment:
I'm not sure why this *was* working, but I'm pretty sure it was not actually integrating the API wrappers and related networking code. This mocks inatjs responses to make sure that when those parts of the app get exercised they get reasonable responses.

Reply:
Does it make sense to have those mocks in the inaturalistjs package itself? Would other clients benefit if we have a shared mock?

Reply:
Not sure. Here the mock implementation is very specific to the test, like we want it to return very specific values. If inatjs were to have built-in mocks, they'd probably need to return generic mock data, so in this case we'd still need to make our own implementation. There might be other cases where that would be useful, though.